listenbrainz/webserver/static/js/src/listens/ListenCard.tsx

```diff
@@ -0,0 +1,207 @@
+import * as timeago from "time-ago";
+
+import * as React from "react";
+import { get as _get } from "lodash";
+import MediaQuery from "react-responsive";
+import {
+  faMusic,
+  faHeart,
+  faHeartBroken,
+} from "@fortawesome/free-solid-svg-icons";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import { getArtistLink, getTrackLink, searchForSpotifyTrack } from "../utils";
+import Card from "../components/Card";
+import APIService from "../APIService";
+
+const DEFAULT_COVER_ART_URL = "/static/img/default_cover_art.png";
+
+type ListenCardProps = {
+  apiUrl: string;
+  listen: Listen;
+  mode: ListensListMode;
+  className?: string;
+  playListen: (listen: Listen) => void;
+};
+
+type ListenCardState = {
+  coverArtUrl?: string;
+  feedback: ListenFeedBack;
+};
+
+export default class ListenCard extends React.Component<
+  ListenCardProps,
+  ListenCardState
+> {
+  private APIService: APIService;
+  playListen: (listen: Listen) => void;
+
+  constructor(props: ListenCardProps) {
+    super(props);
+
+    this.playListen = props.playListen.bind(this, props.listen);
+
+    this.state = {
+      feedback: 0,
+      coverArtUrl: "",
+    };
+
+    this.APIService = new APIService(
+      props.apiUrl || `${window.location.origin}/1`
+    );
+  }
+
+  componentDidMount(): void {
+    this.fetchCoverArt();
+  }
+
+  fetchCoverArt = async () => {
+    const { listen } = this.props;
+    const releaseMBID = _get(
+      listen,
+      "track_metadata.additional_info.release_mbid"
+    );
+    const recordingMSID = _get(
+      listen,
+      "track_metadata.additional_info.recording_msid"
+    );
+
+    this.APIService.getCoverArt(releaseMBID, recordingMSID).then((imageUrl) => {
+      const coverArtUrl = imageUrl !== "" ? "imageUrl" : DEFAULT_COVER_ART_URL;
```

Comment:
you're setting the coverArtUrl to the string `imageUrl` which seems wrong? Have you tested this?

Reply:
Yes this was wrong. Will rebase on API branch and test